sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -371,7 +371,8 @@ object CreateStruct {
       // We should always use the last part of the column name (`c` in the above example) as the
       // alias name inside CreateNamedStruct.
       case (u: UnresolvedAttribute, _) => Seq(Literal(u.nameParts.last), u)
-      case (u @ UnresolvedExtractValue(_, e: Literal), _) => Seq(e, u)
+      // SPARK-43522: make sure the name always be a string literal.
+      case (u @ UnresolvedExtractValue(_, e: Literal), _) => Seq(Literal(e.value.toString), u)
```

Comment:
Do we need to check `if e.resolved` here?

Reply:
Unnecessary, `Literal::resolved` alway be true.